src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/DateOnlyConverter.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class DateOnlyConverter : JsonConverter<DateOnly>
+    {
+        private const int DateOnlyIsoFormatLength = 10; // YYYY-MM-DD
+
+        public override DateOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            DateTime dateTime = reader.GetDateTime();
+            return DateOnly.FromDateTime(dateTime);
```

Comment:
>return DateOnly.FromDateTime(dateTime) [](http://example.com/codeflow?start=12&length=38)

What is the expected behavior here when the DateTime has a time parts? this implementation suggest truncation to Date only. I am wondering if this is a conscious decision? 

In other word, if someone serialized DateTime and try to deserialize it as DateOnly or TimeOnly, should this be allowed?

Reply:
It was deliberate, the idea was to tolerate the trailing time component, much how DateTime ISO parsing tolerates trailing offset data. After feedback from @steveharter and @eerhardt I've pushed an update that restricts the supported formats to calendar dates only.